lib/Collection.js

```diff
@@ -160,6 +160,9 @@ class Collection {
 			} else {
 				this.lastModified = Date.now();
 				this.unindexResource(previous);
+				if (this.Class.prototype.onDelete) {
```

Comment:
Can you change this into `if (typeof previous.onDelete === 'function') {` to test? We don't really care where in the prototype chain it exists. The same comment applies to the block below around line 190.